### PR TITLE
libfaiss: update to 1.7.3

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Update plus use path-style dependency, so that OpenBLAS-devel can be used with it.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.6.8 Server
Xcode 3.2.6

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Review by healchow — APPROVED** [review]

File: math/libfaiss/Portfile [hunk 1]
```diff
@@ -47,7 +47,7 @@ configure.args-append \
                     -DFAISS_ENABLE_PYTHON:BOOL=OFF
 
 variant openblas description {Use OpenBLAS instead of Apple's Accelerate framework} {
-    depends_lib-append  port:OpenBLAS
+    depends_lib-append  path:lib/libopenblas.dylib:OpenBLAS
```
Comment (reneeotten) [inline math/libfaiss/Portfile:50]: I *think* you'd need to increase the "revision" for this change to make sure that the dependencies are correctly reflected in the registry.

Comment (barracuda156) [reply math/libfaiss/Portfile:50]: Ok, will rebase today, thank you.

Comment (barracuda156) [reply math/libfaiss/Portfile:50]: @reneeotten Ah, but we update the version here in this PR from 1.7.2 to 1.7.3. Revbump still needed?

Comment (reneeotten) [reply math/libfaiss/Portfile:50]: sorry, I missed that somehow...

**Pull request merged**